sql/08-is_table_versioned.sql

```diff
@@ -1,28 +1,63 @@
 
+CREATE OR REPLACE FUNCTION _ver_is_table_versioned(
+    p_schema NAME,
+    p_table  NAME
+)
+RETURNS BOOLEAN AS
+$$
+    SELECT EXISTS (
+      SELECT
+          versioned
+      FROM
+          @extschema@.versioned_tables
+      WHERE
+          schema_name = p_schema AND
+          table_name = p_table
+    )
+$$ LANGUAGE sql STABLE;
+
+-- {
 CREATE OR REPLACE FUNCTION ver_is_table_versioned(
     p_schema NAME,
     p_table  NAME
 )
-RETURNS BOOLEAN AS 
+RETURNS BOOLEAN AS
 $$
 DECLARE
     v_is_versioned BOOLEAN;
 BEGIN
-    SELECT
-        versioned
-    INTO
-        v_is_versioned
-    FROM 
-        @extschema@.versioned_tables 
-    WHERE
-        schema_name = p_schema AND
-        table_name = p_table;
+    v_is_versioned := @extschema@._ver_is_table_versioned(p_schema, p_table);
+
+    IF v_is_versioned IS FALSE THEN
+        RETURN FALSE;
+    END IF;
+
+    -- Check that table exists and triggers are in place;
+    -- warn otherwise, see:
+    -- https://github.com/linz/postgresql-tableversion/issues/57
+    BEGIN
+      v_is_versioned := EXISTS (
+        SELECT *
+      FROM pg_trigger t
+      WHERE t.tgrelid = (quote_ident(p_schema) || '.' || quote_ident(p_table) )::regclass
+        AND t.tgname = p_schema || '_' || p_table || '_revision_trg'
+      );
+    EXCEPTION
+      WHEN UNDEFINED_TABLE THEN
+        RAISE WARNING 'Table %.% does not exist', p_schema, p_table;
+        RETURN FALSE;
+      WHEN OTHERS THEN
+        RAISE EXCEPTION 'Got % (%)', SQLERRM, SQLSTATE;
+    END;
 
-    IF v_is_versioned IS NULL THEN
-        v_is_versioned := FALSE;
+    IF NOT v_is_versioned THEN
+        RAISE WARNING 'Table %.% is known as versioned but lacks versioning triggers',
+          p_schema, p_table
+        USING HINT = 'use ver_create_version_trigger to recover';
```

Comment:
I didn't know about `ver_create_version_trigger`, it isn't covered in the [documentation](https://github.com/linz/postgresql-tableversion/blob/master/doc/table_version.md). It would seem cleaner to me that `ver_enable_versioning` is adjusted to fix this situation and suggested as the hint, as you've suggested [here](https://github.com/linz/postgresql-tableversion/issues/57). It is the usual interface as shown in the docs and doesn't require a primary key column parameter.